### DISCUSSION TITLE
More logging to troubleshoot unexpected error in S3 multipart upload

### DISCUSSION THREAD

**Discussion by ruchirj** [duckdb/duckdb#5820] [General]
I am running into this specific error with my local minikube based minio setup, however there is no way to determine what exact error? I have verified there is no issue with IAM/permissions. Some logging before line 267 giving more information about the error would help.

https://github.com/duckdb/duckdb/blob/36f637a40a112c3633c1ecd69420b723ef434a31/extension/httpfs/s3fs.cpp#L267

**Comment by Tishj** [comment]
I also ran into this today, we could just print the raw response in case of error.
It's definitely not pretty and the response could be relatively big in some cases, but I guess when you're troubleshooting you want all of the information you can get

**Comment by avyfain** [comment]
Our team ran into this today as well. The issue we're seeing is likely related to memory management problems in 0.9.2.

More logs are always better
